hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/auth/AuthenticationService.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.cmdgw.auth;
+
+import java.net.URI;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Destroyed;
+import javax.enterprise.context.Initialized;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.websocket.CloseReason;
+import javax.websocket.CloseReason.CloseCodes;
+import javax.websocket.Session;
+
+import org.hawkular.cmdgw.api.AuthMessage;
+import org.hawkular.cmdgw.command.ws.WsEndpoints;
+import org.hawkular.cmdgw.command.ws.server.HttpRequestData;
+import org.hawkular.cmdgw.log.GatewayLoggers;
+import org.hawkular.cmdgw.log.MsgLogger;
+
+import com.squareup.okhttp.ConnectionPool;
+import com.squareup.okhttp.Credentials;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Response;
+
+/**
+ * A service that periodically re-authenticates all open {@link Session}s by forwarding their latest known credentials
```

Comment:
`forwarding their latest known credentials` -- this sounds scary. We should *not* store any credentials in memory, ever.